src/components/Goods/GoodsList.tsx

```diff
@@ -11,18 +11,74 @@ import { IGoodsListProps } from './Types';
 
 import { SORT_GOODS, SORT_TRANSLATE } from '../../constants/sortGoods';
 
+const GOODS_ON_PAGE = 12;
+const AVAILABLE_PAGES = 3;
+
 const GoodsList: React.FC<IGoodsListProps> = ({ filter }) => {
   const [sortSelect, setSortSelect] = useState<keyof typeof SORT_TRANSLATE>(
     'default'
   );
   const { data: goods } = useCockpit<ICockpitGoodsEntries>(true, filter);
-  const { locale } = useRouter();
+  const router = useRouter();
+  const { locale, query } = router;
+  const [currentPage, setCurrentPage] = useState(0);
+  const [visiblePages, setVisiblePages] = useState<Array<number>>([]);
+  const [firstGoodsOnPageIndex, setFirstGoodsOnPageIndex] = useState(0);
+
   const filteredGoods = goods?.entries;
 
   const sortedGoods = SORT_GOODS[sortSelect](
     filteredGoods as ICockpitGoodsRaw[]
   );
 
+  const maxPageNumber = Math.ceil(sortedGoods?.length / GOODS_ON_PAGE);
+
+  useEffect(() => {
+    setFirstGoodsOnPageIndex(currentPage * GOODS_ON_PAGE);
+    if (maxPageNumber <= 1) setVisiblePages([]);
+    else if (currentPage + 1 + AVAILABLE_PAGES >= maxPageNumber)
+      setVisiblePages(
+        [...Array(maxPageNumber).keys()].splice(
+          maxPageNumber >= AVAILABLE_PAGES
+            ? maxPageNumber - AVAILABLE_PAGES
+            : 0,
+          maxPageNumber
+        )
+      );
+    else if (currentPage + 1 - AVAILABLE_PAGES <= 0)
+      setVisiblePages(
+        [...Array(maxPageNumber).keys()].splice(0, AVAILABLE_PAGES)
+      );
+    else
+      setVisiblePages(
+        [...Array(maxPageNumber).keys()].splice(
+          currentPage - Math.ceil(AVAILABLE_PAGES / 2),
+          currentPage + Math.ceil(AVAILABLE_PAGES / 2) - 1
+        )
+      );
+  }, [currentPage, maxPageNumber]);
+
+  useEffect(() => {
+    const page = query.page ? Number.parseInt(query.page as string) : 0;
+    setCurrentPage(page && page < maxPageNumber ? page : 0);
+  }, [query.page, maxPageNumber]);
+
+  const changePage = (pageNum: number) => {
+    const newQuery = router.query;
+    if (pageNum === 0) {
+      delete newQuery.page;
```

Comment:
why do u need to delete page, u can set empty string in query in this line https://github.com/DnCDevelopment/blind-blind/pull/22/commits/b8cffc5f2efcf2308f5773ad792628c4f212ea40#diff-f4b7676bbc1c4ef3b2dd41f0cab363fbaf762a9815848a89c6e85a644d6629deR74